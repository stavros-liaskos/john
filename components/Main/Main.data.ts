import { MainProps } from './Main.types';
import { listData } from '../List/List.data';

const props: MainProps = {
  className: 'mb-auto min-h-[calc(100vh_-_6rem)]',
  i18n: { todo: 'toso' },
  defaultList: listData,
};

export default props;
