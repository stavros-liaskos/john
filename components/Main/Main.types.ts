import { ListEl } from '../List/List.types';

export interface MainI18n {
  todo: string;
}

export interface MainProps {
  i18n: MainI18n;
  className: string;
  defaultList?: ListEl[];
}
