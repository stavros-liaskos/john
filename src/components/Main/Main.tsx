import FollowedArtistList from '../FollowedArtistList/FollowedArtistList';
import React, { Suspense } from 'react';
import Search from '../Search/Search';
import { useUser } from '@auth0/nextjs-auth0/client';
import Login from '../Login/Login';
import Scrapers from '../Scrape/Scrapers';
import Recommendations from '../Recommendations/Recommendations';
import Loading from '../Loading/Loading';
import { searchI18n, followedArtistListI18n, loginI18n, recommendationsI18n } from '../../i18n';

const Main: React.FunctionComponent = () => {
  const { user } = useUser();

  function loginBE() {
    window.location.assign('https://api.releaseraccoon.online/me');
  }

  function fetchy() {
    fetch('https://api.releaseraccoon.online/me', {
      headers: { 'X-Requested-With': 'JavaScript' },
    });
  }

  return (
    <main className="rr-column flex-auto">
      <div className="flex flex-col flex-auto w-full lg:w-9/12">
        <button className={'rr-text btn'} onClick={loginBE}>
          login with assign
        </button>

        <button className={'rr-text btn'} onClick={fetchy}>
          login with fetch
        </button>

        <a href="https://api.releaseraccoon.online/me">
          <button className={'rr-text btn'}>login with anchor</button>
        </a>

        {user ? (
          <>
            <Search i18n={searchI18n} />
            <Scrapers />
            <div className="flex flex-auto flex-col h24">
              <FollowedArtistList i18n={followedArtistListI18n} />
              <Suspense fallback={<Loading />}>
                <Recommendations i18n={recommendationsI18n} />
              </Suspense>
            </div>
          </>
        ) : (
          <Login i18n={loginI18n} />
        )}
      </div>
    </main>
  );
};
Main.whyDidYouRender = false;
export default Main;
