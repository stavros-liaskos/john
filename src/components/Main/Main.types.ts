export interface MainI18n {
  todo: string;
}

export interface MainProps {
  i18n: MainI18n;
}
