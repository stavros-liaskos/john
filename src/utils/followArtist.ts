import { components } from '../types/schema';
import Endpoints from '../types/endpoints';

async function followArtist(artist: components['schemas']['SearchResultArtistDto'], cb: () => void) {
  const headers = new Headers({ 'Content-Type': 'application/json' });
  await fetch(Endpoints.Follow, {
    method: 'POST',
    credentials: 'include',
    headers,
    body: JSON.stringify(artist),
  })
    .then(() => {
      console.log(`${artist.name} followed successfully`);
    })
    .catch(error => {
      console.error('Error:', JSON.stringify(error));
    })
    .finally(() => {
      cb();
    });
}
export default followArtist;
