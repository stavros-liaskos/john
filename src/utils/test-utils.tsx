import React from 'react';
import { render, act } from '@testing-library/react';
import ThemeProvider from '../contexts/Theme/ThemeProvider';
import { UserProvider } from '@auth0/nextjs-auth0/client';
import ArtistsListProvider from '../contexts/ArtistsList/ArtistsListProvider';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ThemeProvider>
      <UserProvider>
        <ArtistsListProvider>{children}</ArtistsListProvider>
      </UserProvider>
    </ThemeProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: { [key: string]: unknown }) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };

export const renderWithAct = async (ui: React.ReactElement, options?: { [key: string]: unknown }) => {
  const component = render(ui, { wrapper: AllTheProviders, ...options });

  const promise = Promise.resolve();
  await act(() => promise);

  return component;
};

export const beforeEachTest = () => {
  jest.resetModules();
  jest.clearAllMocks();
  jest.restoreAllMocks();
  localStorage.clear();
};
