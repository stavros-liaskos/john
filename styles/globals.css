/* purgecss start ignore */
@tailwind base;
@tailwind components;
/* purgecss end ignore */
@tailwind utilities;

@layer components {
  .rr-border {
    @apply dark:border-gh-border border-black;
  }
  .rr-text {
    @apply dark:text-slate-300;
  }
  .h3 {
    @apply rr-text mb-4;
  }
  .btn {
    @apply font-semibold text-sm rounded-sm shadow border-2 rr-border rr-text flex items-center justify-center;
  }
  .btn-large {
    @apply py-2 px-3;
  }
  .btn-small {
    @apply py-1 px-3;
  }
}
