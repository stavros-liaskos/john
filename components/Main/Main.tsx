import { MainProps } from './Main.types';
import List from '../List/List';
import React, { useState } from 'react';
import { ListEl } from '../List/List.types';
import Search from '../Search/Search';
import { searchI18n } from '../Search/Search.data';
import { listI18n } from '../List/List.data';
import DarkMode from '../DarkMode/DarkMode';
import mockedResponse from '../../mocks/searchResult.json';
import { useUser } from '@auth0/nextjs-auth0';
import Login from '../Login/Login';

const Main: React.FunctionComponent<MainProps> = ({ i18n, className, defaultList = [] }) => {
  const { user, error, isLoading } = useUser();

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const [list, setList] = useState<ListEl[]>(
    !defaultList.length ? mockedResponse.artistsPerResource.fromLastfm : defaultList,
  ); // todo rm hardcoded data, get scrapped from db

  if (!i18n || !i18n.todo) {
    return null;
  }

  if (isLoading) return <div>Loading...</div>;
  if (error) return <div>{error.message}</div>;

  return (
    <main className={className}>
      <div className="flex justify-between">
        <DarkMode />
        {user && (
          <button
            className="btn btn-small lg:ml-8 mx-3 my-2"
            onClick={() => (window.location.href = '/api/auth/logout')}
          >
            Logout
          </button>
        )}
      </div>

      {user ? (
        <>
          <Search i18n={searchI18n} />
          <List list={list} i18n={listI18n} />
        </>
      ) : (
        <Login
          i18n={{
            welcome: 'Welcome to Release Raccoon!',
            loginBtn: 'Register',
            text: "Receive your favorite artists' music in your email every week!",
          }}
          handleRegister={() => (window.location.href = '/api/auth/login')}
        />
      )}
    </main>
  );
};

export default Main;
