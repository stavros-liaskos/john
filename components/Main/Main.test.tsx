import React from 'react';
import { beforeEachTest, renderWithAct } from '../../utils/test-utils';
import Main from './Main';
import props from './Main.data';
import { expect } from '@jest/globals';

describe('Main', () => {
  beforeEach(() => beforeEachTest());

  it('renders without data without crashing', async () => {
    // @ts-ignore
    await renderWithAct(<Main />);
  });

  it('show List of results ', async () => {
    const { container } = await renderWithAct(<Main {...props} />);

    // expect(search).toBeTruthy();
    // expect(unfollowBtns.length).toEqual(3);
    expect(container).toMatchSnapshot(); // TODO fix
  });
});
