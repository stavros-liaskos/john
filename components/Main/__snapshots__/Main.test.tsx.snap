// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main show List of results  1`] = `
<div>
  <div>
    invalid json response body at  reason: Unexpected end of JSON input
  </div>
</div>
`;
